clang/lib/CConv/ConstraintResolver.cpp

```diff
@@ -528,6 +541,15 @@ void ConstraintResolver::constrainLocalAssign(Stmt *TSt, DeclaratorDecl *D,
 
   constrainConsVarGeq(V, RHSCons, Info.getConstraints(), PLPtr, CAction, false,
                       &Info);
+
+  if (AllTypes && V.empty() && RHSCons.empty()) {
```

Comment:
Add explanatory comment here, like the one in the other `constrainLocalAssign` above. Also note my comment at that place about the LHS and RHS not necessarily (not likely?) being empty.

Reply:
For example, if you had this code:
```
 int *y =  malloc(sizeof(int)*n);
 int m = n;
```
You would want to keep track of `n` flowing into `m`, but it doesn't appear that this code does that.

Reply:
I think it makes more sense to always have `getExprConstraintVars` return a CV for a base type (we are relying on that behavior for structs) and only return the empty set for NULL. I thought this was the invariant we had now; not sure why it isn't.